src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookMultibranchTrigger.java

```diff
@@ -0,0 +1,110 @@
+import java.util.logging.Logger;
+
+import static java.util.Objects.requireNonNull;
+
+public class BitbucketWebhookMultibranchTrigger extends Trigger<MultiBranchProject<?, ?>> {
+
+    private static final Logger LOGGER = Logger.getLogger(BitbucketWebhookMultibranchTrigger.class.getName());
+
+    @SuppressWarnings("RedundantNoArgConstructor") // Required for Stapler
+    @DataBoundConstructor
+    public BitbucketWebhookMultibranchTrigger() {
+    }
+
+    @Override
+    public BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor getDescriptor() {
+        return (BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor) super.getDescriptor();
+    }
+
+    @Override
+    public void start(MultiBranchProject<?, ?> project, boolean newInstance) {
+        super.start(project, newInstance);
+        if (!newInstance) {
+            return;
+        }
+        BitbucketWebhookMultibranchTrigger.BitbucketWebhookMultibranchTriggerDescriptor descriptor = getDescriptor();
+        project.getSCMSources().stream().filter(scm -> scm instanceof BitbucketSCMSource)
+                .filter(scm -> ((BitbucketSCMSource) scm).isValid())
+                .forEach(scm -> {
+            boolean isAdded = descriptor.addTrigger(project, (BitbucketSCMSource) scm);
+            ((BitbucketSCMSource) scm).setWebhookRegistered(isAdded);
+        });
+    }
+
+    @Symbol("BitbucketWebhookMultibranchTriggerImpl")
+    @Extension
+    public static class BitbucketWebhookMultibranchTriggerDescriptor extends TriggerDescriptor {
+
+        @Inject
+        private BitbucketPluginConfiguration bitbucketPluginConfiguration;
+        @Inject
+        private RetryingWebhookHandler retryingWebhookHandler;
+
+        @SuppressWarnings("unused")
+        public BitbucketWebhookMultibranchTriggerDescriptor() {
+
+        }
+
+        public BitbucketWebhookMultibranchTriggerDescriptor(RetryingWebhookHandler webhookHandler,
+                                                            BitbucketPluginConfiguration bitbucketPluginConfiguration) {
+            this.retryingWebhookHandler = webhookHandler;
+            this.bitbucketPluginConfiguration = bitbucketPluginConfiguration;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return Messages.BitbucketWebhookMultibranchTrigger_displayname();
+        }
+
+        @Override
+        public boolean isApplicable(Item item) {
+            return item instanceof MultiBranchProject;
+        }
+
+        boolean addTrigger(Item item, BitbucketSCMSource scm) {
+            try {
+                registerWebhook(item, scm.getBitbucketSCMRepository());
+                return true;
+            } catch (Exception ex) {
+                LOGGER.log(Level.SEVERE, "There was a problem while trying to add webhook", ex);
+                throw ex;
+            }
+        }
+
+        private BitbucketServerConfiguration getServer(String serverId) {
```

Comment:
I'm just wondering- why does this need to be in the descriptor? I might not understand how triggers work, but it looks like this could be performed from the base class as easily.